docs/source/how-to/custom-formats.md

```diff
@@ -0,0 +1,176 @@
+# Add Support for a Custom Format
+
+For a given "structure family" (e.g. `array`, `dataframe`) the Tiled server
+can provide the data---in whole or in part---in a variety of formats. 
+See {doc}`../tutorials/export` for a list of the formats supported out of the
+box.
+
+To teach Tiled to serve another format, first decide which structure family or
+families are appropriate for this format. For example, it makes sense to export
+a table an Excel spreadsheet, but it does not generally make sense to export an
+N-dimensional array as an Excel spreadsheet. (If you do, you making specific
+assumptions about the array.)
+
+For this guide, we'll take the example of
+[XDI](https://github.com/XraySpectroscopy/XAS-Data-Interchange/blob/master/specification/spec.md#example-xdi-file),
+which is a formalized text-based format for X-ray Spectroscopy data. As you can
+see from the linked example, the file's contents comprise a single table
+and a header of key--value pairs. Therefore, it makes sense to export `dataframe`
+structures and their associated metadata into this format.
+
+Take the following simple server configuration:
+
+```yaml
+# config.yml
+trees:
+- path: /
+  tree: tiled.examples.xas:tree
+```
+
+and serve it:
+
+```
+tiled serve config --public config.yml
+```
+
+As is, we can access the data as CSV, for example.
+
+```
+$ curl -H 'Accept: text/csv' http://localhost:8000/dataframe/full/example
+energy,i0,itrans,mutrans
+8779.0,149013.7,550643.089065,-1.3070486
+8789.0,144864.7,531876.119084,-1.3006104
+8799.0,132978.7,489591.10592,-1.3033816
+8809.0,125444.7,463051.104096,-1.3059724
+8819.0,121324.7,449969.103983,-1.3107085
+8829.0,119447.7,444386.117562,-1.3138152
+8839.0,119100.7,440176.091039,-1.3072055
+8849.0,117707.7,440448.106567,-1.3195882
+8859.0,117754.7,442302.10637,-1.3233895
+8869.0,117428.7,441944.116528,-1.3253521
+8879.0,117383.7,442810.120466,-1.327693
+8889.0,117185.7,443658.11566,-1.3312944
+```
+
+```{note}
+There are three equivalent ways to request a format, more formally called a "media type" or a "MIME type".
+
+    1. Use the standard [HTTP `Accept` Header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept).
+
+    ```
+    $ curl -H 'Accept: text/csv' http://localhost:8000/dataframe/full/example
+    ```
+
+    2. Place the media type in a `format` query parameter.
+
+    ```
+    $ curl -H http://localhost:8000/dataframe/full/example?format=text/csv
```

Comment:
Should be `curl -L ...`

Reply:
Good catch. Was a copy/paste error. Fixed.